data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ParseTreeCoercionException.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+/**
+ * @since 1.3
+ * Exception thrown by {@link ParseTreeCoercionService} methods.
```

Comment:
In line with my other comment on this exception, please update the Javadocs to explain what this means to the caller/client.